modules/core/include/opencv2/core/utils/local_pool.private.hpp

```diff
@@ -0,0 +1,176 @@
+        {
+            const size_t misalign = reinterpret_cast<size_t>(p) % alignment;
+            const size_t adjustment_bytes = misalign ? alignment - misalign : 0;
+            return static_cast<void*>(static_cast<char*>(p) + adjustment_bytes);
+        }
+    public:
+        Block(T *&ptr_, size_t count_, size_t alignment_)
+            : ptr(ptr_), raw_mem(0), count(count_), alignment(alignment_)
+        {
+            CV_Assert(ptr == NULL);
+        }
+        ~Block()
+        {
+            ptr = 0;
+            if (raw_mem)
+                delete[] raw_mem;
+        }
+        size_t getSize() const CV_OVERRIDE
+        {
+            return sizeof(T) * (count + reserve_count());
+        }
+        void real_allocate() CV_OVERRIDE
+        {
+            CV_Assert(ptr == NULL);
+            const size_t full_count = count + reserve_count();
+            raw_mem = new T[full_count];
+            ptr = align_ptr(raw_mem);
+            CV_Assert(reinterpret_cast<size_t>(ptr) % sizeof(T) == 0);
+            CV_Assert(reinterpret_cast<size_t>(ptr) % alignment == 0);
+        }
+        void * fast_allocate(void * buf) const CV_OVERRIDE
+        {
+            CV_Assert(ptr == NULL);
+            buf = align_ptr(buf);
+            CV_Assert(reinterpret_cast<size_t>(buf) % sizeof(T) == 0);
+            CV_Assert(reinterpret_cast<size_t>(buf) % alignment == 0);
+            ptr = static_cast<T*>(buf);
+            return ptr + count;
+        }
+    private:
+        T * &ptr;
+        T * raw_mem;
+        size_t count;
+        size_t alignment;
+    };
+
+public:
+    /** @brief Class constructor.
+
+    @param safe_ Enable _safe_ operation mode, each allocation will be performed independently.
+    */
+    LocalPool(bool safe_ = false);
+
+    /** @brief Class destructor
+
+    All allocated memory well be freed. Each bound pointer will be reset to NULL.
+    */
+    ~LocalPool();
+
+    /** @brief Bind a pointer to local pool.
+
+    LocalPool will store reference to the pointer and allocation parameters effectively owning the
+    pointer and allocated memory. This operation has the same parameters and does the same job
+    as the operator `new`, except allocation can be performed later during the LocalPool::commit call.
+
+    @param ptr Reference to a pointer of type T. Must be NULL
+    @param count Count of objects to be allocated, it has the same meaning as in the operator `new`.
+    @param alignment Alignment of allocated memory. same meaning as in the operator `new` (C++17).
+                     Must be divisible by sizeof(T).
+
+    @note In safe mode allocation will be performed immediatly.
+    */
+    template <typename T>
+    void allocate(T*&ptr, size_t count, size_t alignment = sizeof(T))
+    {
+        CV_Assert(ptr == NULL);
+        CV_Assert(count > 0);
+        CV_Assert(alignment > 0);
+        CV_Assert(alignment % sizeof(T) == 0);
+        blocks.push_back(new Block<T>(ptr, count, alignment));
```

Comment:
> new

Main purpose of original approach is to call "new" only once.
Current approach calls "new" several times in all modes.